types/types.d.ts

```diff
@@ -48,13 +48,13 @@ declare namespace FastifySessionPlugin {
     set(
       sessionId: string,
       session: Fastify.Session,
-      callback: (err: Error) => void
+      callback: (err?: Error) => void
```

Comment:
According to this [typescript Do's and Don'ts handbook section](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#optional-parameters-in-callbacks) callback parameters should not be optional as "it’s always legal to provide a callback that accepts fewer arguments".

But when looking at `express-session` typescript definitions, it's bloated with optional callbacks arguments and that makes it difficult to ensure compatibility between these packages when using typescript :exploding_head:

TBH I chose to follow the good practices of typescript here knowing that it could be controversial after looking at DefinitelyTyped types definitions for `express-session` and its compatible stores. :thinking:

Reply:
I made it optional to match the implementation

Reply:
`?: Error` do not really means optional parameter in this case, but it indicate that `error` may be `undefined`.
Generally speak, it should use `Error | undefined`, but `?: Error` is fewer character and look better.